File: c++/1차 1급 6_initial_code.cpp
// You may use include as below.
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int solution(string pos) {
    int row = pos[1] - '1';
    int col = pos[0] - 'A';
    int dir_rows[8] = { -1, -2, -2, -1, 1, 2, 2, 1 };
    int dir_cols[8] = { -2, -1, 1, 2, -2, -1, 1, 2 };
    int answer = 0;
    
    for (int i = 0; i < 8; i++) {
        int next_row = row + dir_rows[i];
        int next_col = col + dir_cols[i];
        
        if ((0 <= next_row && next_row < 8) && (0 <= next_col && next_col < 8)) {
            answer++;
        }
    }
    
    return answer;
}

// The following is main function to output testcase.
int main() {
    string pos = "A7";
    int ret = solution(pos);

    // Press Run button to receive output. 
    cout << "Solution: return value of the function is " << ret << " ." << endl;
}

File: python/1차 1급 6_initial_code.py
#You may use import as below.
#import math

def solution(pos):
    row = ord(pos[1]) - ord("1")
    col = ord(pos[0]) - ord("A")
    dir_rows = [-1, -2, -2, -1, 1, 2, 2, 1]
    dir_cols = [-2, -1, 1, 2, -2, -1, 1, 2]
    answer = 0

    for i in range(8):
        next_row = row + dir_rows[i]
        next_col = col + dir_cols[i]

        if next_row in range(0, 8) and next_col in range(0, 8):
            answer += 1

    return answer

#The following is code to output testcase.
pos = "A7"
ret = solution(pos)

#Press Run button to receive output.
print("Solution: return value of the function is", ret, ".")